configure.ac

```diff
@@ -5950,10 +5950,16 @@ fi
 # browsers. Only the peer certificate must validate to a trusted
 # certificate. Without this, all certificates sent by a peer must be
 # used in the trust chain or the connection will be rejected.
+
+# We want to turn on alt cert chains by default. This is a convenient feature
+# that would cut down on a lot of support tickets if on by default. But this
+# is heavy feature so not good for the embedded use case. Thus, this default
+# enabled in configure.ac but not in user_settings.h.
+
 AC_ARG_ENABLE([altcertchains],
-    [AS_HELP_STRING([--enable-altcertchains],[Enable using alternative certificate chains, only require leaf certificate to validate to trust root (default: disabled)])],
+    [AS_HELP_STRING([--enable-altcertchains],[Enable using alternative certificate chains, only require leaf certificate to validate to trust root (default: enabled)])],
```

Comment:
Only enable by default if opensslextra or opensslall. `if test "x$ENABLED_OPENSSLALL" = "xyes" || test "x$ENABLED_OPENSSLEXTRA" = "xyes"`.
Also it should be added in `enable-all`. `test "$enable_altcertchains" = "" && enable_altcertchains=yes`

Reply:
Ack...right...I remember that discussion now.  Ok.  In that case, should this be done in settings.h?

Reply:
Not sure. I would start with configure.ac only for now.

Reply:
Turns out I didn't need to check for enable-all. enable-all brings in openssl compat  or it gets brought in some other way.